### ISSUE TITLE
FIx travis ITs

### ISSUE THREAD

**Issue by KostyaSha** [docker-java/docker-java#634]
@alexec  https://travis-ci.org/docker-java/docker-java/builds/144099837 
For tcp connection two tests unstable and sometimes pass, if it order of stdout/stderr then test should be just fixed.

For socket based connection - reproduced always: copyfile(?) and eventteststream(?).

**Comment by alexec** [comment]
Some failing tests:

* com.github.dockerjava.netty.exec.StopContainerCmdExecTest
* com.github.dockerjava.netty.exec.AuthCmdExecTest
* com.github.dockerjava.netty.exec.ExecStartCmdExecTest
* com.github.dockerjava.core.command.StopContainerCmdImplTest
* com.github.dockerjava.core.command.CopyArchiveToContainerCmdImplTest
* com.github.dockerjava.core.command.EventsCmdImplTest
* com.github.dockerjava.netty.exec.AttachContainerCmdExecTest

**Comment by alexec** [comment]
Hi @docker-java/team-docker-java,

Can I please ask what outcome we are after here? For example, is the goal to have stable and useful ITs? I see that the TravisCI builds take 30 minutes and are flaky (random tests fails). Have you thought about using CircleCI instead? If so, why did you decide to run ITs on Travis instead? 

Alex

**Comment by KostyaSha** [comment]
@alexec test fails also on local env, what is the difference where unstable or failing tests?

**Comment by KostyaSha** [comment]
PS. you also is in @docker-java/team-docker-java

**Comment by KostyaSha** [comment]
Yes, the goal of course to have working tests. I already fixed some number of really bad tests + they exposed real bugs.

**Comment by alexec** [comment]
Ah, I didn't realise they fail locally to. That's useful, should be able to fix those independently of the failing builds.

I think of an unstable/flaky test as one that sometimes passes and sometimes fails. 

In case this is an issue with TravisCI, I'm going to try and get it running on another CI. I'll try CircleCI first, but I think they have limits to build times.

**Comment by KostyaSha** [comment]
You should of course try reproduce it locally. For testing different versions i made some dind images
`docker run --privileged -d -p 4444 -e PORT=4444 kostyasha/dind:fedora23-1.11` allows to run on beta for docker. unix socket connection is more difficult as you need to have linux env around mvn execution. But should be easily to run mvn with paused java debug.

**Comment by KostyaSha** [comment]
Having circleci would be also good, more tests - better result.

**Comment by KostyaSha** [comment]
@alexec wiped your accidental commits, rebase if your fork was on top of them.

**Comment by KostyaSha** [comment]
@alexec hi, do you have any luck?

**Comment by KostyaSha** [comment]
@alexec i fixed accidental run of Integration tests on 'test' phase in master.

**Comment by KostyaSha** [comment]
@marcuslinke  https://travis-ci.org/docker-java/docker-java/jobs/146507812
Single failed test reproduced here, it happens only on docker 1.11 tcp connection. `hello world` presents in responce.

**Comment by KostyaSha** [comment]
Reproduced locally and pushed fix to master (sure that it right so without PR).

  - [x] com.github.dockerjava.netty.exec.AttachContainerCmdExecTest

**Comment by marcuslinke** [comment]
@KostyaSha Could you link the appropriate commit here please?

**Comment by KostyaSha** [comment]
> @KostyaSha Could you link the appropriate commit here please?

https://github.com/docker-java/docker-java/commit/4db9bee87c58e564479ed55a4929f3e1fa4662a1

**Comment by KostyaSha** [comment]
Also i limited thread concurrency because it were running upto 14 tests in parallel.

https://travis-ci.org/docker-java/docker-java/builds/146536908
Now tcp connection for 1.11 is green.
Other fixes are welcome!

**Comment by alexec** [comment]
I'd written up some notes about this, I thought I'd added them here. I'll find some time to look at this weekend.

Can I please ask what the five different builds are?

**Comment by KostyaSha** [comment]
@alexec  see `.travis.yml` for options

**Comment by alexec** [comment]
Ah - got it. Matrix builds. Clever stuff.

**Comment by alexec** [comment]
So.... 

https://github.com/docker-java/docker-java/blob/master/.travis.yml#L23

Build are in the same order as the matrix.

**Comment by alexec** [comment]
I will now index the broken tests.

**Comment by alexec** [comment]
| Builds | Failing Test |
| 3 | CopyArchiveToContainerCmdImplTest |
| 3 | EventsCmdImplTest |
| 3 | StatsCmdImplTest |
| 4,5 | AuthCmdImplTest |
| 4,5 | StopContainerCmdExecTest |
| 4 | EventsCmdImplTest |

**Comment by alexec** [comment]
@KostyaSha  - of these failing tests, do you think there's some that needs to be prioritised above others?

**Comment by KostyaSha** [comment]
@alexec no idea, it may happen that they fails only in test code and not in real prods. Try fix the most easiest to limit number of total errors.

**Comment by KostyaSha** [comment]
Fixed more integration tests for 1.12. On 1.12 tcp only one test fails.

**Comment by padyx** [comment]
I analyzed one of the tests failing on master. It's an educated guess by browsing and analyzing the code - but it may help you with fixing it.

**AuthCmdExecTest.testAuthInvalid**: I believe this fails because in docker 1.10.3, there still was an "e-mail" field for the authentication. (E-Mail was removed in 1.11 with PR https://github.com/docker/engine-api/pull/105 and https://github.com/docker/docker/pull/2056 ).  
Since 552b05d by @marcuslinke , the e-mail is not contained in the `docker-java.properties` file anymore. The test will expect an`UnauthorizedException` because the username/password combo should not match. But it looks like the missing e-mail is validated first in 1.10, leading to a different exception.
